wayland-protocols/ext-lockscreen_v1.xml

```diff
@@ -0,0 +1,104 @@
+    may have been unlocked via some other mechanism (e.g. loginctl). Exactly how and
+    when the compositor locks and unlocks is the policy of the compositor.
+
+    A typical use case is for the compositor to lock or unlock when it receives a
+    signal from Logind to do so.
+
+    The key words "must", "must not", "required", "shall", "shall not",
+    "should", "should not", "recommended",  "may", and "optional" in this
+    document are to be interpreted as described in IETF RFC 2119.
+  </description>
+
+  <interface name="ext_lockscreen_manager_v1" version="1">
+    <description summary="used to create lockscreen surfaces">
+      
+    </description>
+    <request name="associate">
+      This request creates a lockscreen surface and asks the compositor to show this
+      surface on the provided output when the session is next locked. If the session is
+      already locked, the compositor will display this surface right away.
+
+      Surfaces may display different content on different outputs.
+      
+      This surface will be shown immediately after the session is locked. It is expected
+      that the compositor does not render intermediary content if a lockscreen surface
+      is provided for an output. If a surface is NOT provided for an output, then it is up to the
+      policy of the compositor to render what it wants.
+
+      The client is expected to destroy lockscreen surfaces as outputs are removed.
+      The client is expected to associate lockscreen surfaces as outputs are added.
+
+      Associating more than one surface for an output is a duplicate_surface error.
+      Associating a surface with a null buffer is a null_buffer error.
+      
+      <arg name="id" type="new_id" interface="ext_lockscreen_surface_v1"/>
+      <arg name="surface" type="object" interface="wl_surface"/>
+      <arg name="output" type="object" interface="wl_output"/>
+    </request>
+
+    <enum name="error">
+      <entry name="duplicate_surface" value="0"
+             summary="attempted to associate more than once surface as the lockscreen surface for an output"/>
+      <entry name="null_buffer" value="1"
+        summary="surface committed with a null buffer"/>
+    </enum>
+  </interface>
+
+  <interface name="ext_lockscreen_surface_v1" version="1">
+    <request name="destroy" type="destructor">
+      This informs the compositor that the lockscreen surface object will no longer be
+      used.
+
+      If the session is locked when this happens, the compositor will display a default
+      render of its choosing until the session is unlocked.
+    </request>
+    
+    <event name="locked">
+      This event is sent when the compositor decides to lock the session.
+      At this time, the compositor will show the surface.
+    </event>
+    <event name="unlocked">
+      This event is sent when the compositor decides to unlock the session.
+      At this time, the compositor will hide the surface.
+    </event>
```

Comment:
Yeah it's useful to show _something_ interesting when a lock first happens, like an animation.

Although maybe "unlock" is less useful, because the surface would have already been hidden at that point. Or maybe we can wait on the lockscreen client's confirmation so that it can first show an unlocking animation and _then_ unlock the session? Maybe a `confirm_unlock` request?